viewing_party/party.py

```diff
@@ -0,0 +1,171 @@
+#WAVE 2
+def get_watched_avg_rating(user_data):
+    list_of_user_watched_movies = user_data["watched"]
+    score = 0
+    average_rating = 0.0
+
+    if list_of_user_watched_movies == []:
+        return average_rating 
+    for movie in list_of_user_watched_movies:
+        score += movie["rating"]
+    average_rating = score/len(list_of_user_watched_movies)
+    return average_rating
+    
+
+def get_most_watched_genre(user_data):
+    list_of_user_watched_movies = user_data["watched"]
+    list_of_genres = []
+
+    if list_of_user_watched_movies == []:
+            return None
+    
+    for movie in list_of_user_watched_movies:
+        list_of_genres.append(movie["genre"])
+
+    popular_genre = max(list_of_genres,key=list_of_genres.count)
+    return popular_genre
+
+
+#WAVE 3
+def get_unique_watched(user_data):
+    list_of_user_watched_movies = user_data["watched"]
+    friend_movie_list = []
+    unique_movie_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movie_list.append(movie) 
+    for movie in list_of_user_watched_movies:
+        if movie not in friend_movie_list:
+            unique_movie_list.append(movie)
+
+    return unique_movie_list 
+
+
+def get_friends_unique_watched(user_data):
+    list_of_user_watched_movies = user_data["watched"]
+    friends_movie_list = []
+    friends_unique_movie_list = []
+    final_friends_unique_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_movie_list.append(movie) 
+
+    for movie in friends_movie_list:
+        if not movie in list_of_user_watched_movies:
+            friends_unique_movie_list.append(movie)
+        
+    for movie in friends_unique_movie_list:
+        if not movie in final_friends_unique_list:
+            final_friends_unique_list.append(movie)
+
+    return final_friends_unique_list
+
+
+#WAVE 4 
+def get_available_recs(user_data):
+    list_of_user_watched_movies = user_data["watched"]
+    
+    list_of_user_subscriptions = user_data["subscriptions"]
+   
+    list_of_rec_movies = []
+
+    friends_movie_list = []
+    user_not_watched_list = []
+    friends_no_dup_unique_movie_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_movie_list.append(movie) 
+    
+    for movie in friends_movie_list:
+        if not movie in friends_no_dup_unique_movie_list:
+            friends_no_dup_unique_movie_list.append(movie)
+    
+    for movie in friends_no_dup_unique_movie_list:
+        watched = False
+        for movies_for_user in list_of_user_watched_movies:
+            if movies_for_user["title"] == movie["title"]:
+                watched = True
+        if watched == False:
+            user_not_watched_list.append(movie)
+
+    for movie in user_not_watched_list:
+        if movie["host"] in list_of_user_subscriptions:
+            list_of_rec_movies.append(movie)     
```

Comment:
there are a lot checks here you can add these checks to for loops you've already created so for example these lines could become
```suggestion
  for movie in friends_watched:
        if movie["host"] in user_data["subscriptions"]:
            recommendations.append(movie)
```

Reply:
Thank you for the suggestions. I had a hard time with this one. I wanted to implement for previous functions but I couldn't get them to work.